Subarray sum

```diff
@@ -0,0 +1,209 @@
+You have been given an unsorted array A of N non negative integers, find a continous sub array which adds to given number S.
+
+
+Input Format:
+
+
+The first line of input contains the integers N and S, where N is size of array and S is the sum.
+
+The second line contains N space seperated integers which are the array elements.
+
+
+
+
+
+Output Format:
+
+
+Print the starting and ending index of the subarray in format " Sum found between index 1 and index 4".
+
+
+
+If no such subarray exist print "No subarray found"
+
+
+
+Constraints:
+
+
+1 <= N <= 10⁷
+
+1 <= Ai <= 10¹⁰
+
+
+
+Example:
+
+
+
+Input 1:
+
+4 7
+
+1 2 3 2
+
+
+
+Ouptut 1:
+
+Sum found between index 1 and index 3
+
+
+
+Explanation 1:
+
+As 2+3+2 = 7 so from index 1 to index 3 elements sum up to form 7.
+
+
+
+Input 2:
+
+3 8
+
+4 5 1
+
+
+
+Output 2:
+
+No such subarray foun
+
+
+
+Explanation 2:
+
+No subarray of elements sum up to give 8.
+
+As 4+5=9, 4+5+1=10, 5+1=6, none matches 8.
+
+
+
+Solution:
+
+
+#include <iostream>
+
+using namespace std;
+
+ 
+
+/* Returns true if the there is a subarray of 
+
+arr[] with a sum equal to 'sum' otherwise 
+
+returns false. Also, prints the result */
+
+int subArraySum(int arr[], int n, int sum)
+
+{
+
+    /* Initialize curr_sum as value of 
+
+    first element and starting point as 0 */
+
+    int curr_sum = arr[0], start = 0, i;
+
+ 
+
+    /* Add elements one by one to curr_sum and 
+
+    if the curr_sum exceeds the sum,
+
+    then remove starting element */
+
+    for (i = 1; i <= n; i++)
+
+    {
+
+        // If curr_sum exceeds the sum,
+
+        // then remove the starting elements
+
+        while (curr_sum > sum && start < i - 1)
+
+        {
+
+            curr_sum = curr_sum - arr[start];
+
+            start++;
+
+        }
+
+ 
+
+        // If curr_sum becomes equal to sum,
+
+        // then return true
+
+        if (curr_sum == sum)
+
+        { 
+
+            cout << "Sum found between indexes "
+
+                 << start << " and " << i - 1;
+
+            return 1;
+
+        }
+
+ 
+
+        // Add this element to curr_sum
+
+        if (i < n)
+
+        curr_sum = curr_sum + arr[i];
+
+    }
+
+ 
+
+    // If we reach here, then no subarray
+
+    cout << "No subarray found";
+
+    return 0;
+
+}
+
+   // start of main function
+
+int main()
+
+{
+
+    int n, sum;
+
+    cin>>n>>sum;
+
+    int arr[n];
+
+    for (int y=0;y<n;y++)
+
+        {
+
+        cin>>arr[y];
+
+        }
+
+    subArraySum(arr, n, sum);
+
+    return 0;
+
+}
+
+
+
+Solution Explanation:
+
```

Comment:
Too much explanation, the code should be explanatory. Please remove all the  explanation that are not needed